// cpanel-api/js/modules/ZoneEdit/index.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var abstract_1 = require("../abstract");
var ZoneEdit = (function (_super) {
    __extends(ZoneEdit, _super);
    function ZoneEdit() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    ZoneEdit.prototype.fetchzone_records = function (profile, options, callback) {
        var _this = this;
        this.getActionSlug(profile, {
            version: 2, module: 'ZoneEdit', func: 'fetchzone_records'
        }, function (err, action) {
            _this.call(profile, {
                params: options,
                action: action
            }, callback);
        });
    };
    ZoneEdit.prototype.add_zone_record = function (profile, options, callback) {
        var _this = this;
        this.getActionSlug(profile, {
            version: 2, module: 'ZoneEdit', func: 'add_zone_record'
        }, function (err, action) {
            _this.call(profile, {
                params: options,
                action: action
            }, callback);
        });
    };
    return ZoneEdit;
}(abstract_1.cPanelModule));
exports.ZoneEdit = ZoneEdit;
