// cpanel-api/js/modules/abstract.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var request = require("request");
var cPanelModule = (function () {
    function cPanelModule() {
    }
    cPanelModule.prototype.call = function (profile, options, callback) {
        var params = options.params, action = options.action;
        var port = action.split('/').shift();
        action = action.slice(5);
        var data = {
            baseUrl: 'https://' + profile.host + port,
            uri: '',
            headers: {
                'Authorization': 'Basic ' + new Buffer(profile.user + ':' + profile.pass).toString('base64')
            },
            form: {}
        };
        this.getAuthToken(profile, data, function (err, res) {
            data.form = params;
            data.uri = res + action;
            request.get(data, function (err, response) {
                if (err)
                    return callback(new Error(err), response);
                var res = response.toJSON();
                if (res.statusCode !== 200)
                    return callback(new Error(res.body), response);
                try {
                    var body = JSON.parse(res.body);
                    return callback(err, body);
                }
                catch (e) {
                    return callback(new TypeError('Response was not JSON'), res);
                }
            });
        });
    };
    cPanelModule.prototype.getAuthToken = function (profile, data, callback) {
        data.form = {
            user: profile.user,
            pass: profile.pass
        };
        data.uri = 'login';
        var req = request(data, function (err, res) {
            if (err)
                callback(err, null);
            else
                callback(null, res.req.path.split('/')[1]);
        });
    };
    cPanelModule.prototype.getActionSlug = function (profile, options, callback) {
        var version = options.version, module = options.module, func = options.func;
        switch (version) {
            case 2: {
                var action = ':2083/json-api/cpanel?cpanel_jsonapi_user=' + profile.user +
                    '&cpanel_jsonapi_apiversion=2&cpanel_jsonapi_module=' + module +
                    '&cpanel_jsonapi_func=' + func;
                return callback(null, action);
            }
            case 'UAPI': {
                var action = ':2083/execute/' + module + '/' + func + '?';
                return callback(null, action);
            }
            case 'WHM': {
                var action = ':2087/json-api/' + func + '?api.version=1';
                return callback(null, action);
            }
        }
    };
    return cPanelModule;
}());
exports.cPanelModule = cPanelModule;
