// cpanel-api/js/modules/Email/index.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var abstract_1 = require("../abstract");
var Email = (function (_super) {
    __extends(Email, _super);
    function Email() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    Email.prototype.addpop = function (profile, options, callback) {
        var _this = this;
        if (!options.quota)
            options.quota = 0;
        this.getActionSlug(profile, {
            version: 2, module: 'Email', func: 'addpop'
        }, function (err, action) {
            _this.call(profile, {
                params: options,
                action: action
            }, callback);
        });
    };
    Email.prototype.add_pop = function (profile, options, callback) {
        var _this = this;
        if (!options.quota)
            options.quota = 0;
        this.getActionSlug(profile, {
            version: 'UAPI', module: 'Email', func: 'add_pop'
        }, function (err, action) {
            _this.call(profile, {
                params: options,
                action: action
            }, function (err, res) {
                if (err)
                    return callback(err, res);
                if (res.errors)
                    return callback(new Error(res.errors), res);
                if (res.status && res.status !== 1)
                    return callback(new Error(res.data), res);
                else
                    return callback(null, res.data);
            });
        });
    };
    Email.prototype.add_mx = function (profile, options, callback) {
        var _this = this;
        options.alwaysaccept = options.alwaysaccept || 1;
        this.getActionSlug(profile, {
            version: 'UAPI', module: 'Email', func: 'add_mx'
        }, function (err, action) {
            _this.call(profile, {
                params: options,
                action: action
            }, callback);
        });
    };
    Email.prototype.add_forwarder = function (profile, options, callback) {
        var _this = this;
        if (!options.fwdopt)
            options.fwdopt = 'fwd';
        this.getActionSlug(profile, {
            version: 'UAPI', module: 'Email', func: 'add_forwarder'
        }, function (err, action) {
            _this.call(profile, {
                params: options,
                action: action
            }, callback);
        });
    };
    return Email;
}(abstract_1.cPanelModule));
exports.Email = Email;
